app/services/requirement_form_json_service.rb

```diff
@@ -340,6 +342,91 @@ def get_multi_contact_datagrid_form_json(requirement_block_key = requirement&.re
     }
   end
 
+  def get_nested_info_component(field_key, parent_key, label, required, field_type = nil)
+    key = snake_to_camel(field_key.to_s)
+    key = "#{parent_key}|#{key}" if parent_key.present?
+
+    if field_type.present?
+      raise Error if DEFAULT_FORMIO_TYPE_TO_OPTIONS[field_type.to_sym].blank?
+      {
+        key: key,
+        type: field_type,
+        requirementInputType: field_type,
+        tableView: true,
+        input: true,
+        label: label,
+        widget: {
+          type: "input",
+        },
+      }.merge!(DEFAULT_FORMIO_TYPE_TO_OPTIONS[field_type.to_sym])
+    else
+      {
+        applyMaskOn: "change",
+        tableView: true,
+        input: true,
+        key: key,
+        label: label,
+        type: "textfield",
+        validate: {
+          required: required,
+        },
+        **DEFAULT_FORMIO_TYPE_TO_OPTIONS[:text],
+      }
+    end
+  end
+
+  def get_pid_info_components(requirement_block_key = requirement&.requirement_block&.key)
+    return {} unless requirement.input_type_pid_info?
+    # key = "#{requirement.key(requirement_block_key)}|pid_info?"
+    component = {
+      legend: requirement.label,
+      key: requirement_block_key,
+      type: "fieldset",
+      custom_class: "contact-field-set",
```

Comment:
I think this class is supposed to be specific for contacts. If not needed, remove